pkg/master/client_ca_hook.go

```diff
@@ -58,8 +58,14 @@ func (h ClientCARegistrationHook) PostStartHook(hookContext genericapiserver.Pos
 		return h.tryToWriteClientCAs(client)
 	})
 
-	// if we're never able to make it through initialization, kill the API server
-	if err != nil {
+	switch err {
+	case nil:
+	case wait.ErrWaitTimeout:
```

Comment:
this clears the signal that something failed on startup (/healthz goes green)... how would someone know there was an issue to fix?

Reply:
In this implementation the error in the log would be the only way to know

Reply:
@deads2k thoughts on this?

Reply:
Could we remember the last error from within the poll and then additional log it as well as the timeout? Doesn't entirely address Jordan's concern but would provide some additional feedback.

Reply:
That sounds like a good idea